nixos/modules/system/boot/networkd.nix

```diff
@@ -468,6 +469,19 @@ let
         (assertMinimum "Table" 0)
       ];
 
+      sectionWLAN = checkUnitConfig "WLAN" [
+        (assertOnlyFields [
+          "PhysicalDevice"
+          "Type"
+          "WDS"
+        ])
+        # systemd supports strings here too ("phy0") but index is good enough
```

Comment:
Is that confusing for people?

Reply:
I removed the type check and added a comment.